### DISCUSSION TITLE
Issues starting the Docker version

### DISCUSSION THREAD

**Discussion by jbogdani** [maplibre/martin#1280] [Q&A] [answered]
Hello, 
I'm trying to run martin on MacOS Sonoma 14.4.1, by following the guide provided.
I have tried to run 

```bash
export PGPASSWORD=postgres
docker run \
  --net=host  \
  -p 3000:3000 
  -e PGPASSWORD \
  -e DATABASE_URL=postgresql://postgres@localhost:5432/localgis \
  ghcr.io/maplibre/martin
```
all I get is the following error:
```txt
Unable to get a Postgres connection from the pool localgis: Error occurred while creating a new object: error connecting to server: Address not available (os error 99)
```

I have also installed martin using [homebrew](https://brew.sh/) and it is working perfectly:

```bash
martin postgresql://postgres:postgres@localhost:5432/localgis
```

Of course, I can provide some more logs... if properly instructed :)

**Comment by nyurik** [comment]
You are trying to connect to PostgreSQL running on localhost from inside the docker. In theory, the `--net host` should allow for that. In order to debug, try to use standard `psql` command to connect, something like this:

```
docker run \
  --rm -it \
  --net=host  \
  -e DATABASE_URL=postgresql://postgres@localhost:5432/localgis \
  postgres bash
```

and run  `psql $DATABASE_URL`.

If I remember correctly, Martin doesn't actually use `PGPASSWORD` - that's a `psql` feature.  So if it works on the host (via Homebrew), you might be using some other way of authentication.

**Comment by jbogdani** [comment]
Hello, thank you for the reply.
When I run
```bash
docker run \
  --rm -it \
  --net=host  \
  -e DATABASE_URL=postgresql://postgres:postgres@localhost:5432/localgis \
  postgres bash
```
and then 
```bash
psql $DATABASE_URL
```
I get the following error:
```
psql: error: connection to server at "localhost" (::1), port 5432 failed: Connection refused
	Is the server running on that host and accepting TCP/IP connections?
connection to server at "localhost" (127.0.0.1), port 5432 failed: Connection refused
	Is the server running on that host and accepting TCP/IP connections?
```

Is this a docker-related issue?

**Reply by nyurik** [reply, marked as answer]
It might be that your postgres DB is configured to refuse connections from anywhere except the localhost, and I suspect that connecting from inside docker is treated as other machine somehow?  Not certain
